docs/glossary.asciidoc

```diff
@@ -0,0 +1,32 @@
+[id="{p}-glossary"]
+== Glossary
+
+This glossary covers Elastic Cloud on Kubernetes. If a term is not in this list, you can also look in the link:https://kubernetes.io/docs/reference/glossary/[Kubernetes glossary].
+
+[[GKE]]GKE::
+  Google Kubernetes Engine (GKE): is a managed, production-ready environment for deploying containerized applications.
+
+[[AKS]]AKS::
+  Azure Kubernetes Service (AKS): using Kubernetes on Microsoft Azure to manage your hosted Kubernetes environment.
+
+[[EKS]]EKS::
+  Elastic Kubernetes Service (Amazon EKS): using Kubernetes on AWS to deploy, manage, and scale containerized applications.
+
+[[ECK]]ECK::
+  Elastic Cloud on Kubernetes (ECK) extends the basic Kubernetes orchestration capabilities to deploy and manage Elasticsearch, Kibana and APM Server.
+  
+[[k8s]]k8s::
+  Abbreviation derived by replacing the 8 letters “ubernete” with “8”.
```

Comment:
I would word this as:

Numeronym for "Kubernetes" derived by replacing "ubernete" with "8".

Reply:
I feel that if someone knows what numeronym is then 'derived by...' part is not needed. If someone doesn't know then the definition is not helpful as one needs to check what numeronym is. I think I'd merge these and use both abbreviation and numeronym.

Reply:
Should we keep it simple and just leave "abbreviation"?